L1Trigger/L1THGCal/src/be_algorithms/HGCalShowerShape.cc

```diff
@@ -294,18 +300,27 @@ float HGCalShowerShape::sigmaPhiPhiMax(const l1t::HGCalMulticluster& c3d) const
 
 float HGCalShowerShape::eMax(const l1t::HGCalMulticluster& c3d) const {
   
-   const edm::PtrVector<l1t::HGCalCluster>& clustersPtrs = c3d.constituents();
+  std::unordered_map<int, float> layer_energy;
 
-   float EMax=0;   
+  const edm::PtrVector<l1t::HGCalCluster>& clustersPtrs = c3d.constituents();
+
+  for(const auto& clu : clustersPtrs){
+    
+    int layer = HGC_layer(clu->subdetId(),clu->layer());        
+    layer_energy[layer] += clu->energy();
 
-   for(const auto& clu : clustersPtrs){
+  }
+  
+  float EMax=0;   
+  
+  for(const auto& layer : layer_energy){
 
-     if(clu->energy()>EMax)
-       EMax = clu->energy();
+     if(layer.second>EMax)
```

Comment:
Try to use consistent indentations within a file as much as possible. 3 spaces are used here instead of 2 in other places, and 4 spaces were used in the previous version of the file.